test/puppeteer.js

```diff
@@ -0,0 +1,135 @@
+let browser = null;
+// Launch the browser and open a new blank page
+describe("jobs-ejs puppeteer test", function () {
+    before(async function () {
+        this.timeout(10000);
+        //await sleeper(5000)
+        browser = await puppeteer.launch({ headless: false, slowMo: 10 });
+        page = await browser.newPage();
+        await page.goto("http://localhost:3000");
+    });
+    after(async function () {
+        this.timeout(5000);
+        await browser.close();
+    });
+    describe("got to site", function () {
+        it("should have completed a connection", async function () { });
+    });
+    describe("index page test", function () {
+        this.timeout(10000);
+        it("finds the index page logon link", async () => {
+            this.logonLink = await page.waitForSelector(
+                "a ::-p-text(Click this link to logon)",
+            );
+        });
+        it("gets to the logon page", async () => {
+            await this.logonLink.click();
+            await page.waitForNavigation();
+            const email = await page.waitForSelector('input[name="email"]');
+        });
+    });
+    describe("logon page test", function () {
+        this.timeout(20000);
+        it("resolves all the fields", async () => {
+            this.email = await page.waitForSelector('input[name="email"]');
+            this.password = await page.waitForSelector('input[name="password"]');
+            this.submit = await page.waitForSelector("button ::-p-text(Logon)");
+        });
+        it("sends the logon", async () => {
+            testUser = await seed_db();
+            await this.email.type(testUser.email);
+            await this.password.type(testUserPassword);
+            await this.submit.click();
+            await page.waitForNavigation();
+            await page.waitForSelector(`p ::-p-text(${testUser.name} is logged on.)`);
+            // await page.waitForSelector("a ::-p-text(change the secret)");
+            await page.waitForSelector('a[href="/secretWord"]');
+            const copyr = await page.waitForSelector("p ::-p-text(copyright)");
+            const copyrText = await copyr.evaluate((el) => el.textContent);
+            console.log("copyright text: ", copyrText);
+        });
+
+    });
+
+    describe("puppeteer crud operations", function () {
+        this.timeout(15000);
+        it("clicks the link to access tires list", async () => {
+
+            const { expect } = await import('chai');
+
+            // selector for the link
+            // this.tiresLink = await page.waitForSelector('a[href="/secretWord"]');
+            this.tiresLink = await page.waitForSelector("a ::-p-text(Placeholder)");
+
+
+            await this.tiresLink.click();
+            await page.waitForNavigation();
+
+            const content = await page.content();
+            const pageParts = content.split("<tr>");
+
+            // assignment said to expect 20 but I am getting 21 not sure if I am wrong somewhere
```

Comment:
Have you checked what’s in the seeded db file? And/or if you’ve implemented a paging limit